fix(federation): convert query graph node's type for fetch dependency graph path

duckki opened the pull request:
Fetch dependency graph expects supergraph types.
So, Interface object types from subgraphs need to be converted to interface type positions.

Also, added `--type-conditioned-fetching` option to the CLI tool.

<!-- ROUTER-924 -->
---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [x] Manual Tests

svc-apollo-docs commented:
### 🛠️ Docs Preview Building...

<details>
<summary>Configuration</summary>

```json
{
  "repoOverrides": {
    "apollographql/router@dev": {
      "remote": {
        "owner": "apollographql",
        "repo": "router",
        "branch": "duckki/router-924"
      }
    }
  }
}
```

</details>

900b7d3c7fd15ecb449e030ecc1d3b78dec1ec4d

sachindshinde reviewed (commented):
The third commit https://github.com/apollographql/router/pull/6433/commits/900b7d3c7fd15ecb449e030ecc1d3b78dec1ec4d looks fine. The first two commits are from https://github.com/apollographql/router/pull/6431 , where there's some pending feedback. Once that PR is updated/merged and this PR rebased, this PR should be good to go.

duckki commented:
The first two commits are not necessary for this fix. I rebased it on `origin/dev` and dropped the other two commits, so this PR can be reviewed independently.
